src/lib.rs

```diff
@@ -490,29 +501,31 @@ pub trait PwmAccess<T> {
 impl<DV: DeviceVariant, I: RegisterAccess> PwmAccess<u16> for Lp586x<DV, I, DataMode16Bit> {
     type Error = I::Error;
 
-    fn set_pwm(&mut self, start: u16, values: &[u16]) -> Result<(), Self::Error> {
-        let mut registers = [0; 396];
+    fn set_pwm(&mut self, start_dot: u16, values: &[u16]) -> Result<(), Self::Error> {
+        let mut buffer = [0; Variant0::NUM_DOTS as usize * 2];
```

Comment:
We should document why we are wasting so much stack space here:
```suggestion
        // rust does not support making this dependent on a generic parameter (DV::NUM_DOTS),
        // so have have to just use the biggest variant here. Lots of stack space wasted in
        // most cases :(
        let mut buffer = [0; Variant0::NUM_DOTS as usize * 2];
```

I don't really have a better idea at the moment at least...

Reply:
Yes.. agree this is a bit wasteful, i also found you cannot use the generic here, so this seems the safest/easiest way to go for now..  (isn't this functionality hidden behind a 'const_generics' feature flag of some sort...?)

Reply:
I don't think this particular issues is hidden behind any feature flag, at least to my limited knowledge. I also couldn't find a workaround, but haven't searched for too long. It'd be best if we wouldn't need a buffer at all, but at least `embedded_hal` v1 removes the the `_iter` methods and I'm not sure if they were even implemented for most platforms.

One alternative might be using a const generic in the function signature, i.e. `fn set_pwm<const N: usize>(&mut self, start_dot: u16, values: &[u16; N])`, but that probably has it's implications as well (e.g. you can't just use `let foo = [0; {N*2}]` without the nightly feature `generic_const_exprs`).

Reply:
I think the waste is limited. When on a (very) resource constrained platform, you can always revert to the 8-bit mode (also implemented in this PR) that doesn't not create an extra intermediate buffer. On the other hand, whilst the `*Iter` methods are nice, for instance on my platform (nrf) i2c is done using DMA, for which a buffer in RAM is needed in any case (I don't see how this iter approach would work in the DMA world..)
Lochily more modern platforms typically have some kb's of memory to spare..